### DISCUSSION TITLE
RDKit Molecule Slide Generator - "Advanced MolsToGridImage"

### DISCUSSION THREAD

**Discussion by kienerj** [rdkit/rdkit#3652] [General]
Dear RDKit community,

as a long time user of RDKit and also silent reader on the mailing list, time has come to give something back to the community. Please see my new personal project [RDKit Molecule Slide Generator](https://github.com/kienerj/molecule-slide-generator). What a mouthful. As the saying goes there are only 2 hard things in computer science: cache invalidation and naming things.

The readme in the linked GitHub repository should explain in depth what Slide-Generator is about. In a nutshell it's an "advanced" `MolsToGridImage`. The "advanced" part being that you can add multiple styled text properties below each molecule and not just 1 line. But images say more than words:

![Example Slide](https://raw.githubusercontent.com/kienerj/molecule-slide-generator/master/images/example_slide_bzr.png)

Yes, a possible 2D alignment you would have to apply beforehand and yes, for unknown reason I really like this comic-type font. But that is also a feature. You can choose any TrueType font.

Slide-Generator also makes use of RDKits new PNG metadata feature. What does it mean? **The chemical structure and the displayed properties are stored inside the images metadata and can be extracted again**. Therefore if you share this image on a blog, via email or put it inside an MS office document, the displayed data is shared too! The `DataExtractor` extracts the molecule and properties into a pandas data frame for further processing.

If you are interested in Slide-Generator please go ahead and try it out, provide feedback (in the repos issues) or fork it and adjust it to your personal needs. Happy coding,

:santa: :christmas_tree: Happy holidays! :christmas_tree: :santa:

**Comment by greglandrum** [comment]
That is so cool! Thanks Joos!

**Reply by bp-kelley** [reply]
This is awesome.  It's coming at the right time for me for sure!
